test/test_response_header.rb

```diff
@@ -29,7 +30,7 @@ def server_run(app: @app, early_hints: false)
   end
 
   def send_http_and_read(req)
-    send_http(req).read
+    send_http(req).sysread 10_240
```

Comment:
🤔

Reply:
`read` is evil.  I was working on this, and with `read`, the new test took 20 sec.  With `sysread`, it was what it should be.

But, then the Windows tests (only Ruby 3.1 and later), weren't getting the full response.  But, using PR #2896, no problem.

Some of the issue is using the same Process for the server and client...

Reply:
I see 👍 Lets refactor the magic number into a constant somwhere then.

Reply:
Done.  Added comment also, as the value is dependent on the tests.  I've used looping code for responses with unknown length...